jupyterhub/orm.py

```diff
@@ -163,28 +163,32 @@ def __repr__(self):
 # lots of things have roles
 # mapping tables are the same for all of them
 
-_role_map_tables = []
+_role_associations = {}
 
-for has_role in (
+for entity in (
     'user',
     'group',
     'service',
 ):
-    role_map = Table(
-        f'{has_role}_role_map',
+    table = Table(
+        f'{entity}_role_map',
         Base.metadata,
         Column(
-            f'{has_role}_id',
-            ForeignKey(f'{has_role}s.id', ondelete='CASCADE'),
+            f'{entity}_id',
+            ForeignKey(f'{entity}s.id', ondelete='CASCADE'),
             primary_key=True,
         ),
         Column(
             'role_id',
             ForeignKey('roles.id', ondelete='CASCADE'),
             primary_key=True,
         ),
+        Column('managed_by_auth', Boolean, default=False),
```

Comment:
I think this should have `nullable=False` (also in the alembic upgrade, I think). Can it also be checked that after the upgrade, the column is filled with False instead of Null? That could probably be tested in the db test_upgrade.

I can work on that, if you're not comfortable with it.

Reply:
This indeed sounds simple on the surface, but the migration scripts start failing if I toggle nullable.

> I can work on that, if you're not comfortable with it.

I would appreciate that, thank you!

Reply:
No problem at all! I'll have a go.